addon/-microstate.js

```diff
@@ -22,11 +23,25 @@ export default Ember.Helper.extend({
     return decorate(this, actions, this.wrap(this.value), [this.value]);
 
     function decorate(microstate, actions, object, context) {
-      return Object.create(object, Object.keys(actions).reduce((values, key)=> {
-        return assign(values, {
-          [key]: descriptor(valueFor(actions, key))
+      let transitions = reduceObject(actions, (result, name) => {
+        return assign(result, {
+          [name]: descriptor(valueFor(actions, name))
         });
-      }, {}));
+      });
+
+      if (object instanceof Array) {
```

Comment:
Instead of special casing the `Array`, let's put this in the `toHandlebarsValue` method which doesn't exist yet :), but will with #28. There we can make the appropriate values enumerable.